pilot/cmd/pilot-agent/status/server.go

```diff
@@ -714,6 +728,66 @@ func (s *Server) handleAppProbeTCPSocket(w http.ResponseWriter, prober *Prober)
 	}
 }
 
+func (s *Server) handleAppProbeGRPC(w http.ResponseWriter, req *http.Request, prober *Prober) {
+	opts := []grpc.DialOption{
+		grpc.WithBlock(),
+		// nolint: staticcheck
+		grpc.WithInsecure(), // credentials are currently not supported
+	}
+	if userAgent := req.Header["User-Agent"]; len(userAgent) > 0 {
+		// simulate kubelet
+		// please refer to:
+		// https://github.com/kubernetes/kubernetes/blob/v1.23.1/pkg/probe/grpc/grpc.go#L56
+		// https://github.com/kubernetes/kubernetes/blob/v1.23.1/pkg/probe/http/http.go#L103
+		opts = append(opts, grpc.WithUserAgent(userAgent[0]))
+	}
+
+	timeout := time.Duration(prober.TimeoutSeconds) * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	addr := net.JoinHostPort(s.appProbersDestination, fmt.Sprintf("%d", prober.GRPC.Port))
+	conn, err := grpc.DialContext(ctx, addr, opts...)
+	if err != nil {
+		log.Errorf("Failed to create grpc connection to probe app: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	defer conn.Close()
+
+	var svc string
+	if prober.GRPC.Service != nil {
+		svc = *prober.GRPC.Service
+	}
+	grpcClient := grpcHealth.NewHealthClient(conn)
+	resp, err := grpcClient.Check(ctx, &grpcHealth.HealthCheckRequest{
+		Service: svc,
+	})
+	if err != nil {
+		status, ok := grpcStatus.FromError(err)
+		if ok {
+			switch status.Code() {
+			case codes.Unimplemented:
+				log.Errorf("server does not implement the grpc health protocol (grpc.health.v1.Health): %v", err)
+			case codes.DeadlineExceeded:
+				log.Errorf("grpc request not finished within timeout: %v", err)
+			default:
+				log.Errorf("grpc probe failed: %v", err)
+			}
+		} else {
+			log.Errorf("grpc probe failed: %v", err)
+		}
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if resp.GetStatus() == grpcHealth.HealthCheckResponse_SERVING {
+		w.WriteHeader(http.StatusOK)
```

Comment:
return

Reply:
Why is return needed? there's no more instructions below

Reply:
So can get rid of else